### PR TITLE
Big Changes

### PR DESCRIPTION
**Pull request by timgrant**
Tons of changes. Too many too list. Biggest changes were to the sampler constructor, and the addition of numerous sample sets. Made some graph changes to handle issues related to deleting tracks and nodes. Numerous small tweaks and fixes.

### CODE REVIEW

**Comment by timgrant** [comment]
You'll just want to merge this one and dive on in, since it has numerous changes which the current front-end relies on. I'm just leaving this open so you can take a look at it, but it's kinda hard to parse due to all the added files...

**Comment by sphippen** [comment]
Yeah, the changes I saw seemed reasonable. Merging.
